src/simulation/dynamics/constraintEffector/_UnitTest/test_ConstraintDynamicEffectorUnit.py

```diff
@@ -0,0 +1,331 @@
+    r_B2C_N = r_B2N_N - r_CN_N
+    # set initial attitudes to zero (note assumption that B1, B2, and N frames all initially aligned is used later)
+    sigma_B1N = [[0.0], [0.0], [0.0]]
+    sigma_B2N = [[0.0], [0.0], [0.0]]
+    # compute relative velocity due to spin and COM offset
+    target_spin = [0.01,0.01,0.01]
+    omega_CN_N = np.array(target_spin)
+    omega_B1N_B1 = omega_CN_N
+    omega_B2N_B2 = omega_CN_N
+    dv_B1C_N = np.cross(omega_CN_N,r_B1C_N)
+    dv_B2C_N = np.cross(omega_CN_N,r_B2C_N)
+    rDot_B1N_N = rDot_B1N_N + dv_B1C_N
+    rDot_B2N_N = rDot_B2N_N + dv_B2C_N
+
+    # Set the initial values for all spacecraft states
+    scObject1.hub.r_CN_NInit = r_B1N_N
+    scObject1.hub.v_CN_NInit = rDot_B1N_N
+    scObject1.hub.sigma_BNInit = sigma_B1N
+    scObject1.hub.omega_BN_BInit = omega_B1N_B1
+    scObject2.hub.r_CN_NInit = r_B2N_N
+    scObject2.hub.v_CN_NInit = rDot_B2N_N
+    scObject2.hub.sigma_BNInit = sigma_B2N
+    scObject2.hub.omega_BN_BInit = omega_B2N_B2
+
+    # Create the constraint effector module
+    constraintEffector = constraintDynamicEffector.ConstraintDynamicEffector()
+    # Set up the constraint effector
+    constraintEffector.r_P1B1_B1 = r_P1B1_B1
+    constraintEffector.r_P2B2_B2 = r_P2B2_B2
+    constraintEffector.r_P2P1_B1Init = r_P2P1_B1Init
+    constraintEffector.alpha = 1e3
+    constraintEffector.beta = constraintEffector.alpha
+    constraintEffector.ModelTag = "constraintEffector"
+    # Add constraints to both spacecraft
+    scObject1.addDynamicEffector(constraintEffector)
+    scObject2.addDynamicEffector(constraintEffector)
+
+    # Add test module to runtime call list
+    unitTestSim.AddModelToTask(unitTaskName, scObject1)
+    unitTestSim.AddModelToTask(unitTaskName, scObject2)
+    unitTestSim.AddModelToTask(unitTaskName, constraintEffector)
+
+    # Record the spacecraft state message
+    datLog1 = scObject1.scStateOutMsg.recorder()
+    datLog2 = scObject2.scStateOutMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, datLog1)
+    unitTestSim.AddModelToTask(unitTaskName, datLog2)
+
+    # Log energy and momentum variables
+    conservationData1 = scObject1.logger(["totOrbAngMomPntN_N", "totRotAngMomPntC_N", "totOrbEnergy", "totRotEnergy"])
+    conservationData2 = scObject2.logger(["totOrbAngMomPntN_N", "totRotAngMomPntC_N", "totOrbEnergy", "totRotEnergy"])
+    unitTestSim.AddModelToTask(unitTaskName,conservationData1)
+    unitTestSim.AddModelToTask(unitTaskName,conservationData2)
+
+    # Initialize the simulation
+    unitTestSim.InitializeSimulation()
+
+    # Setup and run the simulation
+    stopTime = 1
+    unitTestSim.ConfigureStopTime(macros.sec2nano(stopTime))
+    unitTestSim.ExecuteSimulation()
+
+    # collect the recorded spacecraft states
+    constraintTimeData = datLog1.times() * macros.NANO2SEC
+    r_B1N_N_hist = datLog1.r_BN_N
+    rdot_B1N_N_hist = datLog1.v_BN_N
+    sigma_B1N_hist = datLog1.sigma_BN
+    omega_B1N_B1_hist = datLog1.omega_BN_B
```

Comment:
unused variable, this can be removed?

Reply:
Assigning an initial attitude has now been removed.